x/nexus/keeper/rate_limit.go

```diff
@@ -55,6 +56,20 @@ func (k Keeper) SetRateLimit(ctx sdk.Context, chainName exported.ChainName, limi
 		return fmt.Errorf("%s is not a registered asset for chain %s", limit.Denom, chain.Name)
 	}
 
+	events.Emit(ctx, &types.RateLimitUpdated{
+		Chain:  chain.Name,
+		Limit:  limit,
+		Window: window,
+	})
+
+	// delete any rate limit info if provided limit is max uint256
+	if limit.Amount.Equal(sdk.NewIntFromBigInt(utils.MaxUint.BigInt())) {
+		k.getStore(ctx).DeleteNew(getRateLimitKey(chain.Name, limit.Denom))
```

Comment:
this way we don't show a huge confusing number for assets that don't need a rate limit. It also has the benefit that we don't bother tracking epoch amounts for such assets.